classifier_and_summarizer/summarizer/extractor_output_preprocessor.py

```diff
@@ -0,0 +1,205 @@
+        self.font_style = font_style
+        self.embedding = embedding
+
+
+class ExtractorOutputPreprocessor:
+    ''' Class to implement the utilities for
+    preprocessing the extractor output
+
+    Preprocessing steps applied:
+    * Split contents into different types
+    * Assign img_description_embeddings to media
+    * Summarize the text content
+    '''
+
+    def __init__(self, contents):
+        self.content_list = contents.content_list
+        self.normal_text_content_list = list()  # non-title text
+        self.title_text_content_list = list()  # title text
+        self.media_content_list = list()  # images/gifs
+        self.embedded_content_list = list()  # insta/tweets/quotes
+        self.text_summarizer = TextSummarizer(priority="speed")
+        self.sentence_embedding_model \
+            = SentenceTransformer('bert-base-nli-stsb-mean-tokens')
+
+    def get_preprocessed_content_lists(self):
+
+        # Assign 'index' field to every content
+        for content_index in range(len(self.content_list)):
+            self.content_list[content_index].index = content_index
+
+        #  first split content into different categories
+        self._split_content()
+
+        # summarize the webpage text
+        self._summarize_text_content()
+
+        # assemble the sentence objects
+        self._make_sentence_objects()
+
+        # add image description embeddings
+        self._add_media_description_and_attribute_embeddings()
+
+        return {
+            "titles": self.title_text_content_list,
+            "sentences": self.sentence_objects_list,
+            "media": self.media_content_list,
+            "embedded_content": self.embedded_content_list
+        }
+
+    def _split_content(self):
+        ''' Splits the given content into 4 lists
+
+        * list of title text
+        * list of normal text
+        * list of media content (images/gifs)
+        * list of embedded content
+        '''
+        for content in self.content_list:
+            if content.content_type == ContentType.TEXT:
+                if content.type == "title":
+                    self.title_text_content_list.append(content)
+                else:
+                    self.normal_text_content_list.append(content)
+
+            elif content.content_type == ContentType.IMAGE:
+                self.media_content_list.append(content)
+
+            else:
+                self.embedded_content_list.append(content)
+
+    def _summarize_text_content(self):
+        '''
+        Applies text summarization to the combined text in the webpage
+        '''
+        # aggregate all text from the webpage
+        webpage_text = '. '.join(
+            [text.text_string for text in self.normal_text_content_list])
+        # sentence tokenize to get list of summary sentences
+        self.summarized_text = sent_tokenize(
+            self.text_summarizer.summarize_text(webpage_text))
+
+    def _make_sentence_objects(self):
```

Comment:
Break this method

Reply:
Updated accordingly! - thanks for the suggestion